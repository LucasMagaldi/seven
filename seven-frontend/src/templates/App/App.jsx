import React from "react";
import { Wrapper } from './styles'

function App({word}) {
  return (
    <Wrapper className="App">
      <h1>Home</h1>
      <p data-testid="word">{word}</p>
    </Wrapper>
  );
}

export default App;
