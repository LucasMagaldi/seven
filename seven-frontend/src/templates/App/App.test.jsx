import { render, screen } from '@testing-library/react';
import App from './App';

describe("Testing App", () => {
  it('renders learn react link', () => {
  const { debug } = render(<App />);
  debug();
  const headingContainer = screen.getByRole('heading', {name: 'Home'}).parentElement;
  expect(headingContainer).toHaveStyle({
    background: 'red'
  });

  expect(headingContainer).toMatchSnapshot();

});

  it("Should test Props", () => {
    const { getByTestId } = render(<App word={2}/>);
    const wordValue = getByTestId("word").textContent;
    expect(wordValue).toEqual(7);
  });
})


