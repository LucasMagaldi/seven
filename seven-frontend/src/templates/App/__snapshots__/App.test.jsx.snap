// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App renders learn react link 1`] = `
.c0 {
  background: red;
}

<div
  class="c0 App"
>
  <h1>
    Home
  </h1>
  <p
    data-testid="word"
  />
</div>
`;

exports[`renders learn react link 1`] = `
.c0 {
  background: red;
}

<div
  class="c0 App"
>
  <h1>
    Home
  </h1>
</div>
`;
